hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8265,14 +8265,22 @@ private WALEdit reckonDeltas(Operation op, Mutation mutation, Durability effecti
           break;
         default: throw new UnsupportedOperationException(op.toString());
       }
-      int newCellSize = PrivateCellUtil.estimatedSerializedSizeOf(newCell);
-      if (newCellSize > this.maxCellSize) {
-        String msg = "Cell with size " + newCellSize + " exceeds limit of " + this.maxCellSize
-          + " bytes in region " + this;
+      if (this.maxCellSize > 0) {
+        int newCellSize = PrivateCellUtil.estimatedSerializedSizeOf(newCell);
+        if (newCellSize > this.maxCellSize) {
+          String msg = "Cell with size " + newCellSize + " exceeds limit of " + this.maxCellSize
+            + " bytes in region " + this;
+          if (LOG.isDebugEnabled()) {
+            LOG.debug(msg);
+          }
+          throw new DoNotRetryIOException(msg);
+        }
+      } else {
+        String msg = "Cell size check is disabled because of maxCellSize is set to" +
+          this.maxCellSize;
         if (LOG.isDebugEnabled()) {
           LOG.debug(msg);
```

Comment:
We should use log4j parameterized messages feature, that way we don't need the concatenation logic, nor the if statement.

Reply:
ok